graphdatascience/graph/graph_entity_ops_runner.py

```diff
@@ -85,6 +89,25 @@ def stream(
                 columns={"variable": "nodeProperty", "value": "propertyValue"}
             )
 
+        if db_node_properties:
+            nodeIds = result["nodeId"].tolist()
+            db_nodes = self._util_proc_runner.asNodes(nodeIds)
```

Comment:
this is not very efficient as we get the whole node objects via bolt.

Would be interesting how large the performance impact is, but I would rather use a specific YIELD clause in the cypher query to only return the desired properties instead of returning the whole nodes.

Reply:
A custom query could be something like

```
MATCH (n) WHERE id(n) IN $nodeIds
RETURN n.p1 AS p1, n.p2 AS p2, n.p3 AS p3
```

where `pi` are the names given by parameter